src/composables/trade/useSor.ts

```diff
@@ -245,17 +245,28 @@ export default function useSor({
           tokenOutAddress.toLowerCase()
         );
 
-        const tokenInAmount = BigNumber.from(deltas[tokenInPosition]).abs();
-
-        const tokenOutAmount = BigNumber.from(deltas[tokenOutPosition]).abs();
+        let tokenInAmount = BigNumber.from(deltas[tokenInPosition]).abs();
+        let tokenOutAmount = BigNumber.from(deltas[tokenOutPosition]).abs();
 
         if (swapType === SwapType.SwapExactOut) {
+          tokenInAmount = await mutateAmount({
+            amount: tokenInAmount,
+            address: tokenInAddressInput.value,
+            isTokenIn: true,
+          });
+
           tokenInAmountInput.value = tokenInAmount.gt(0)
             ? formatAmount(formatUnits(tokenInAmount, tokenInDecimals))
             : '';
         }
 
         if (swapType === SwapType.SwapExactIn) {
+          tokenOutAmount = await mutateAmount({
+            amount: tokenOutAmount,
+            address: tokenOutAddressInput.value,
+            isTokenIn: false,
+          });
+
```

Comment:
In this function, if the in or out token is stETH the deltas returned the out token is actually the wstETH amount and so needs to be converted using our mutateAmount function.